test/model_test.py

```diff
@@ -85,3 +88,85 @@ def test_group_cardinality_fully_blocked(self):
         self.db.add(pkg)
         self.db.commit()
         self.assertEqual(0, group.package_count)
+
+
+@patch('sqlalchemy.sql.expression.func.clock_timestamp',
+       return_value=literal_column("'2017-10-10 10:00:00'"))
+class PackagePriorityTest(DBTest):
+    def setUp(self):
+        super(PackagePriorityTest, self).setUp()
+        self.pkg = self.prepare_packages('rnv')[0]
+        self.pkg.resolved = True
+        self.build = self.prepare_build('rnv', state=True)
+        self.build.started = '2017-10-10 10:00:00'
+
+    def get_priority(self, pkg):
+        return self.db.query(
+            Package.current_priority_expression(
+                collection=pkg.collection,
+                last_build=pkg.last_build,
+            )
+        ).filter(Package.id == pkg.id).scalar()
+
+    def get_priority_join(self, pkg):
+        return self.db.query(
+            Package.current_priority_expression(
+                collection=Collection,
+                last_build=Build,
+            )
+        ).join(Package.collection)\
+            .join(Package.last_build)\
+            .filter(Package.id == pkg.id).scalar()
+
+    def assert_priority(self, expected, pkg=None):
+        pkg = pkg or self.pkg
+        self.db.commit()
+        self.assertAlmostEqual(expected, self.get_priority(pkg))
+        self.assertAlmostEqual(expected, self.get_priority_join(pkg))
```

Comment:
It would be good to guard against None operands (if expected is None then assert that priority is None too, else assert they are almost equal). Otherwise you get confusing TypeError when assertion fails.